### PR TITLE
Update metadata dependencies

### PR DESCRIPTION
**Pull request by liorblob**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/25951
fixes: https://github.com/demisto/etc/issues/25441

## Description
Merge user dependencies and generated dependencies 
Add core packs mandatory dependency validation

### CODE REVIEW

File: Tests/Marketplace/marketplace_services.py [hunk 1]
```diff
@@ -1451,3 +1448,16 @@ def get_higher_server_version(current_string_version, compared_content_item, pac
...
     finally:
         return higher_version_result
+
+
+def set_pack_dependencies(pack_name, user_metadata, packs_dependencies_mapping):
+    pack_dependencies = packs_dependencies_mapping.get(pack_name, {}).get('dependencies', {})
+    if pack_name in GCPConfig.CORE_PACKS_LIST:
+        mandatory_dependencies = [dependency for dependency in pack_dependencies
+                                  if dependency.get('mandatory', False) is True]
+        if mandatory_dependencies:
```
Comment (yuvalbenshalom) [inline Tests/Marketplace/marketplace_services.py:1458]: we should raise an exception if there is new mandatory dependency for a core pack if it answers the the terms:
1. the new dependency is not in the core packs group
2. the new dependency is not in the user_metadata dependencies.
3. the dependency is mandatory

only then we should raise an error and fail the build.
and it is better if we will print all the new problematic dependencies found and prompt all the list.

File: Tests/Marketplace/marketplace_services.py [hunk 2]
```diff
@@ -1451,3 +1448,16 @@ def get_higher_server_version(current_string_version, compared_content_item, pac
                     f"Additional info:\n {e}")
     finally:
         return higher_version_result
+
+
+def set_pack_dependencies(pack_name, user_metadata, packs_dependencies_mapping):
```
Comment (IkaDemisto) [inline Tests/Marketplace/marketplace_services.py:1453]: It can be part of Pack class

File: Tests/Marketplace/marketplace_services.py (hunk 1 above)
Comment (IkaDemisto) [reply Tests/Marketplace/marketplace_services.py:1458]: ➕ 1️⃣

Comment (IkaDemisto) [reply Tests/Marketplace/marketplace_services.py:1458]: ➕ 1️⃣

**Comment by liorblob** [comment]
Modified per https://github.com/demisto/etc/issues/26149#issuecomment-653850297

**Pull request merged**